cpp/pipelines/augment_canones_references.cc

```diff
@@ -0,0 +1,210 @@
+/** \file    augment_canones_references.cc
+ *  \brief   A tool for adding numerical canon law references to MARC-21 datasets.
+ *  \author  Dr. Johannes Ruscheinski
+ */
+
+/*
+    Copyright (C) 2019, Library of the University of Tübingen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <cstdlib>
+#include "BibleUtil.h"
+#include "Compiler.h"
+#include "MARC.h"
+#include "StringUtil.h"
+#include "UBTools.h"
+#include "util.h"
+
+
+namespace {
+
+
+void ParseRanges(const std::string &ppn, const std::string &ranges, unsigned * const range_start, unsigned * const range_end) {
```

Comment:
It would be a good idea to write a bit of documentation for the different magic numbers being used. The info on the wiki page doesn't directly translate to the code here (at least at first blush).